### PR TITLE
Testcase success temporary data cleanup refactoring

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

- reduce testreport file to minimal size
-  remove old magic list of cleanup directories and handle cleanup of testdata inside more special testcases.

### CODE REVIEW

File: js/client/modules/@arangodb/testsuites/authentication.js [hunk 1]
```diff
@@ -182,10 +182,6 @@ function authenticationParameters (options) {
 
   let continueTesting = true;
   let results = {};
-  // we append one cleanup directory for the invoking logic...
-  let dummyDir = fs.join(fs.getTempPath(), 'authentication_dummy');
-  fs.makeDirectory(dummyDir);
-  pu.cleanupDBDirectoriesAppend(dummyDir);
```
Comment (adityamukho) [inline js/client/modules/@arangodb/testsuites/authentication.js:188]: Don't we need to create `authentication_dummy` using `tmpDirManager` to replace the deteted code above?

File: js/client/modules/@arangodb/testsuites/dump.js [hunk 2]
```diff
@@ -157,12 +171,11 @@ class DumpRestoreHelper extends tu.runInArangoshRunner {
       this.restoreConfig.setThreads(this.restoreOptions.threads);
     }
     if (this.restoreOptions.jwtSecret) {
-      let keyDir = fs.join(fs.getTempPath(), 'jwtSecrets');
-      if (!fs.exists(keyDir)) {  // needed on win32
-        fs.makeDirectory(keyDir);
+      this.otherKeyDir = fs.join(fs.getTempPath(), 'jwtSecrets');
```
Comment (adityamukho) [inline js/client/modules/@arangodb/testsuites/dump.js:174]: This is the same as `this.keyDir`. Is this intentional? Can we not just reuse `this.keyDir` if they're both the same?

**Review by adityamukho — COMMENTED** [review]
A few minor comments.

File: js/client/modules/@arangodb/testsuites/arangosh.js [hunk 3]
```diff
@@ -98,14 +99,8 @@ function arangosh (options) {
...
-    let weirdNames = ['some dog', 'ла́ять', '犬', 'Kläffer'];
-    let tmpPath = fs.getTempPath();
-    let tmp = fs.join(tmpPath, weirdNames[0], weirdNames[1], weirdNames[2], weirdNames[3]);
-    process.env.TMPDIR = tmp;
-    process.env.TEMP = tmp;
-    process.env.TMP = tmp;
-    fs.makeDirectoryRecursive(process.env.TMPDIR);
-    pu.cleanupDBDirectoriesAppend(tmp);
+    let weirdNames = ['arangosh_tests', 'some dog', 'ла́ять', '犬', 'Kläffer'];
+    let tmpMgr = new tmpDirMmgr(fs.join(...weirdNames), options);
```
[the remark this answers is not in the document]
Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/arangosh.js:103]: make it two arrays?

File: js/client/modules/@arangodb/testsuites/dump.js (hunk 2 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/dump.js:174]: ok, changed.

File: js/client/modules/@arangodb/testsuites/authentication.js (hunk 1 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/authentication.js:188]: this is done by the instance manager inside the loop for each test:
```
    let instanceManager = new im.instanceManager('tcp', options,
                                                 authTestServerParams[test],
                                                 'authentication_parameters_' + authTestNames[test]);
```

**Pull request merged**